src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/FilterSet.cs

```diff
@@ -195,16 +188,6 @@ public ImmutableArray<CompletionFilterWithState> GetFilterStatesInSet(bool addUn
             return builder.ToImmutableAndFree();
         }
 
-        private readonly struct FilterWithMask
-        {
-            public readonly CompletionFilter Filter;
-            public readonly int Mask;
-
-            public FilterWithMask(CompletionFilter filter, int mask)
-            {
-                Filter = filter;
-                Mask = mask;
-            }
-        }
+        private readonly record struct FilterWithMask(CompletionFilter Filter, int Mask);
```

Comment:
please move to separate file.

Reply:
That feels unnecessary to me, given this is a private, one-line auxiliary type. Is this required by our convention?